### PR TITLE
Fixes #7291: Exit sending inventory to rudder-webapp if the queue is full

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/7291

### CODE REVIEW

File: initial-promises/node-server/distributePolicy/1.0/propagatePromises.cf [hunk 1]
```diff
@@ -89,23 +89,23 @@ bundle agent sendInventoryToCmdb
...
-      "${g.rudder_inventories}/accepted-nodes-updates"
+      "${updates_inventory_files}"
         transformer => "${g.rudder_tools}/send-clean.sh http://${rudder_inventory_endpoint_server}:8080/endpoint/upload/ ${this.promiser} ${g.rudder_inventories}/received/ ${g.rudder_inventories}/failed/",
-        depth_search => recurse_visible(1),
-        file_select => all_inventory_files,
         classes => if_else("inventory_sent", "rudder_send_inventory_to_cmdb_cant_send_inventory"),
+        ifvarclass => "!rudder_send_inventory_to_cmdb_cant_send_inventory",
         comment => "Sending the inventory to the cmdb";
 
 # NEED TO DO THE RSYNC PART
```
Comment (amousset) [inline initial-promises/node-server/distributePolicy/1.0/propagatePromises.cf]: Off topic, but do you know what it refers to?

Comment (jooooooon) [inline initial-promises/node-server/distributePolicy/1.0/propagatePromises.cf]: Yeah, it's really old, when we were considering rsync for copying promises from main server to relay servers, IIRC. You can remove it :)

**Comment by jooooooon** [comment]
This looks great! Can you just confirm you have tested it?

**Comment by amousset** [comment]
Tested with broken inventories and with too many inventories:
```
...
2016-04-06T13:00:50+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv246.ocs'[163]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2016-04-06T13:00:50+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv246.ocs'[163]:                                  Dload  Upload   Total   Spent    Left  Speed
100  328k    0    62  100  328k    190  1009k --:--:-- --:--:-- --:--:-- 1012kdder/inventories/incoming/inv246.ocs'[163]: 
2016-04-06T13:00:50+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv246.ocs'[163]: Transformer '/var/rudder/inventories/incoming/inv246.ocs' => '/var/rudder/tools/send-clean.sh http://localhost:8080/endpoint/upload/ /var/rudder/inventories/incoming/inv246.ocs /var/rudder/inventories/received/ /var/rudder/inventories/failed/' seemed to work ok
2016-04-06T13:00:50+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv247.ocs'[164]: Transforming '/var/rudder/tools/send-clean.sh http://localhost:8080/endpoint/upload/ /var/rudder/inventories/incoming/inv247.ocs /var/rudder/inventories/received/ /var/rudder/inventories/failed/' 
2016-04-06T13:00:50+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv247.ocs'[164]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2016-04-06T13:00:50+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv247.ocs'[164]:                                  Dload  Upload   Total   Spent    Left  Speed
100  328k    0    62  100  328k    175   930k --:--:-- --:--:-- --:--:--  962kdder/inventories/incoming/inv247.ocs'[164]: 
2016-04-06T13:00:51+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv247.ocs'[164]: Transformer '/var/rudder/inventories/incoming/inv247.ocs' => '/var/rudder/tools/send-clean.sh http://localhost:8080/endpoint/upload/ /var/rudder/inventories/incoming/inv247.ocs /var/rudder/inventories/received/ /var/rudder/inventories/failed/' seemed to work ok
2016-04-06T13:00:51+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]: Transforming '/var/rudder/tools/send-clean.sh http://localhost:8080/endpoint/upload/ /var/rudder/inventories/incoming/inv248.ocs /var/rudder/inventories/received/ /var/rudder/inventories/failed/' 
2016-04-06T13:00:51+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2016-04-06T13:00:51+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]:                                  Dload  Upload   Total   Spent    Left  Speed
100  328k    0     0  100  328k      0   936k --:--:-- --:--:-- --:--:--  940kdder/inventories/incoming/inv248.ocs'[165]: 
2016-04-06T13:00:51+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]: curl: (22) The requested URL returned error: 503
2016-04-06T13:00:51+0200    error: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]: Finished command related to promiser '/var/rudder/inventories/incoming/inv248.ocs' -- an error occurred, returned 22
2016-04-06T13:00:51+0200    error: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]: Transformer '/var/rudder/inventories/incoming/inv248.ocs' => '/var/rudder/tools/send-clean.sh http://localhost:8080/endpoint/upload/ /var/rudder/inventories/incoming/inv248.ocs /var/rudder/inventories/received/ /var/rudder/inventories/failed/' returned error
2016-04-06T13:00:51+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]: WARNING: Unable to send /var/rudder/inventories/incoming/inv248.ocs, inventory endpoint is temporarily unavailable, will retry later
2016-04-06T13:00:51+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]: This often happens due to rate-throttling in the endpoint to save on memory consumption. This is standard behavior.
2016-04-06T13:00:51+0200     info: /default/sendInventoryToCmdb/files/'/var/rudder/inventories/incoming/inv248.ocs'[165]: Transforming '/var/rudder/tools/send-clean.sh http://localhost:8080/endpoint/upload/ /var/rudder/inventories/incoming/inv248.ocs /var/rudder/inventories/received/ /var/rudder/inventories/failed/' 
error    DistributePolicy          Send inventories to CMDB                     Some inventories failed to add successfully to Rudder
```

**Pull request merged**